### PR TITLE
os/arch/arm/src/amebasmart: Enable/disable loguart rx irq according t…

### PR DESCRIPTION
**Pull request by zhongnuo-tang**
…o driver call

1. Loguart rx irq should be handled properly with enable/disable

### CODE REVIEW

File: os/arch/arm/src/amebasmart/amebasmart_serial.c [hunk 1]
```diff
@@ -717,10 +717,13 @@ static void rtl8730e_log_up_rxint(struct uart_dev_s *dev, bool enable)
...
 	DEBUGASSERT(priv);
 	priv->rxint_enable = enable;
-	//if (enable)
+	if (enable) {
+		LOGUART_INTConfig(LOGUART_DEV, LOGUART_BIT_ERBI, ENABLE);
 		//LOGUART_RxCmd(LOGUART_DEV, ENABLE);
-	//else
+	} else {
+		LOGUART_INTConfig(LOGUART_DEV, LOGUART_BIT_ERBI, DISABLE);
 		//LOGUART_RxCmd(LOGUART_DEV, DISABLE);
```
Comment (ewoodev) [inline os/arch/arm/src/amebasmart/amebasmart_serial.c:725]: why you keep this code??

**Review by ewoodev — COMMENTED** [review]
I would like to check how it is currently going from booting to when LOG UART RX is first enabled in this regard.

if we enable here, do we need to set privious changs of LOG UART enable during boot?

**Comment by ewoodev** [comment]
The UART porting layer of TizenRT remains unchanged from amebaD.
I think that routing the LOG UART through up_lowgetc and up_lowputc has caused a number of issues.
Could you please explain why this structure was changed and why it is being maintained?

**Comment by zhongnuo-tang** [comment]
> I would like to check how it is currently going from booting to when LOG UART RX is first enabled in this regard.
> 
> if we enable here, do we need to set privious changs of LOG UART enable during boot?

Hi @ewoodev, we can remove the enable during boot with this changes.

**Comment by zhongnuo-tang** [comment]
> The UART porting layer of TizenRT remains unchanged from amebaD. I think that routing the LOG UART through up_lowgetc and up_lowputc has caused a number of issues. Could you please explain why this structure was changed and why it is being maintained?

amebaD's loguart is a common UART which can use the same set of structure like UART1.. However, in amebasmart, there is a dedicated UART for LOGUART only, that is why we are using a separate set of flow different from UART0/1/2..
image

File: os/arch/arm/src/amebasmart/amebasmart_serial.c (hunk 1 above)
Comment (zhongnuo-tang) [reply os/arch/arm/src/amebasmart/amebasmart_serial.c]: removed

**Comment by sunghan-chang** [comment]
@zhongnuo-tang Could you let us know some information as below?
1. How did you verify the PR?
2. If we don't apply this, what happen? (I mean what was the issue?)

**Comment by sunghan-chang** [comment]
@ewoodev @gSahitya-samsung @neel-samsung @ritesh55555 This is necessary to review and to verify. Could you check that?

**Comment by ewoodev** [comment]
Hello @zhongnuo-tang 
Could you please verify uart stuck if enter tash during booting??

**Comment by zhongnuo-tang** [comment]
> Hello @zhongnuo-tang Could you please verify uart stuck if enter tash during booting??

verified with loadable_ext DDR, spamming tash during booting, no stuck occurred.

**Review by sunghan-chang — APPROVED** [review]

**Comment by ewoodev** [comment]
> > Hello @zhongnuo-tang Could you please verify uart stuck if enter tash during booting??
> 
> verified with loadable_ext DDR, spamming tash during booting, no stuck occurred.

Could you please verify psram also?

**Comment by zhongnuo-tang** [comment]
> > > Hello @zhongnuo-tang Could you please verify uart stuck if enter tash during booting??
> > 
> > 
> > verified with loadable_ext DDR, spamming tash during booting, no stuck occurred.
> 
> In my test, the stuck is occurred. Could you please verify psram also?

Hi @ewoodev,
Using latest tizenrt sha:511aa520a8c9b080331d60522fe85aa7ba1bfca1
Verified with loadable_ext_psarm: spamming tash during booting, no stuck occurred.
Verified with loadable_ext_ddr: spamming tash during booting, no stuck occurred.
Verified with loadable_apps: spamming tash during booting, no stuck occurred.

Maybe you can share with the git sha u tested on? or which defconfig u are using?

**Review by ewoodev — APPROVED** [review]

**Review by jylee9613 — APPROVED** [review]

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**